py/objstr.c

```diff
@@ -1285,6 +1297,30 @@ mp_obj_t mp_obj_str_format(mp_uint_t n_args, const mp_obj_t *args, mp_map_t *kwa
         }
     }
 
+    if (str <= top) {
+        int count = str - start - 1;
+        if (!level) {
+            count += 1;
+        }
+        vstr_cut_out_bytes(str_input, position, count);
```

Comment:
You mean "insert blank bytes" if insertion is bigger than replaced substring or "cut out" in another case and make memory move after?